Sources/Application/MainCoordinator.swift

```diff
@@ -44,21 +44,25 @@ final class MainCoordinator: NSObject,
         loginCoordinator?.handleUniversalLink(url)
     }
     
-    func evaluateRevisit(action: () -> Void) {
-        if userStore.returningAuthenticatedUser {
-            do {
-                tokenHolder.accessToken = try userStore.secureStoreService.readItem(itemName: .accessToken)
-                homeCoordinator?.updateToken(accessToken: tokenHolder.accessToken)
-                action()
-            } catch {
-                print("Error getting token: \(error)")
+    func evaluateRevisit(action: @escaping () -> Void) {
+        Task {
+            await MainActor.run {
```

Comment:
Wouldn't marking this method (and the other dealing with local auth in the `LoginCoordinator`) as `@MainActor` achieve essentially the same thing without having to include a Task and having the closure be `@escaping`?
There was a pretty sizeable refactor to do away with using `waitForTruth` where it isn't absolutely necessary and avoid flakey tests so would certainly be on the side of that where possible.

Reply:
Still doesn't work.  I admit this is an edge case, as it only seems to happen on my iPad, so it is unlikely to be encountered by other users.  On the other hand, you are pretty much blocked from using the app with local auth if you run into it.